lib/5-years-of-ios.rb

```diff
@@ -0,0 +1,145 @@
+
+# This class could be extracted if we do another timeline.
+class Event 
+  attr_accessor :year
+  attr_accessor :heading
+  attr_accessor :content
+  attr_accessor :image_url
+
+  def initialize (year = nil, heading, content, image_url)
+    @year = year
+    @heading = heading
+    @content = content
+    @image_url = image_url
+  end
+
+  def formatted_date
+    if @year
+      "<div class='cntl-icon cntl-center'>#{@year}</div>"
+    else
+      ""
+    end
+  end
+
+  def formatted_image
+    if @image_url
+      "<div class='cntl-image'><img src='#{ @image_url }'></div>"
+    else
+      ""
+    end
+  end
+
+  def formatted_heading
+    if @heading
+      id = @heading.gsub(/\s/, '-').downcase
+      id.gsub!(/[^a-zA-Z-]/, '')
+      "<h3 id='#{ id }'>#{ @heading }</h3>"
+    else
+      ""
+    end
+  end
+
+  def formatted_content
+    if @content
+      @content.map { |p| "<p>#{ p }</p>\n" }.reduce(:+)
+    else
+      ""
+    end
+  end
+
+  def to_html
+    <<-eos
+      <div class="cntl-state">
+        #{ formatted_date } 
+        <div class="cntl-content">
+          #{ formatted_heading }
+          #{ formatted_content }
+        </div>
+        #{ formatted_image }
+      </div>
+    eos
+  end
+end
+
+module FiveYears
+
+  public
+
+  def self.Events
+    events = []
```

Comment:
this stuff should ideally go in a middleman data file: https://middlemanapp.com/advanced/data_files/